pkg/ingester/client/client.go

```diff
@@ -35,19 +35,11 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet) {
 func New(cfg Config, addr string) (grpc_health_v1.HealthClient, error) {
 	opts := []grpc.DialOption{
 		grpc.WithInsecure(),
-		grpc.WithUnaryInterceptor(grpc_middleware.ChainUnaryClient(
-			otgrpc.OpenTracingClientInterceptor(opentracing.GlobalTracer()),
-			middleware.ClientUserHeaderInterceptor,
-		)),
-		grpc.WithStreamInterceptor(grpc_middleware.ChainStreamClient(
-			otgrpc.OpenTracingStreamClientInterceptor(opentracing.GlobalTracer()),
-			middleware.StreamClientUserHeaderInterceptor,
-		)),
 		grpc.WithDefaultCallOptions(
-			grpc.MaxCallRecvMsgSize(cfg.MaxRecvMsgSize),
 			grpc.UseCompressor("gzip"),
```

Comment:
Wouldn't this be overwritten by `cfg.GRPCClientConfig.DialOption()` to false by default ?

Reply:
A test would make me happy here to prove it.

Reply:
It won't be overwritten. We are setting defaults which can be overwritten while invoking the grpc call.
cfg.GRPCClientConfig.DialOption is also setting default options. https://github.com/grafana/loki/blob/ddddd14726bf0190eb87035e9aab77769f410a0d/vendor/github.com/cortexproject/cortex/pkg/util/grpcclient/grpcclient.go#L57

Reply:
This is actually connection level default which you can overwriting during each grpc call.

Reply:
both set grpc.WithDefaultCallOptions through DialOption. I'm not so sure about this.

Reply:
The option array here has 2 grpc.WithDefaultCallOptions one with gzip one without.

Reply:
Ho I get it ! the second on will just not add the compressor. the first on adds it. ok.